guide/10-mapping-and-visualization-alt/part7_utility_functions_for_the_map_widget.ipynb

```diff
@@ -0,0 +1,357 @@
+{
```

Comment:
<p>Next, we want a user to draw a freehand <code>polyline</code> to indicate the paths they take for their runs. When the drawing operation ends, we use the <code>GIS</code>'s Geometry service to compute the length of the drawn path. We can do this by adding an event listener to the map widget that gets called when drawing is completed (i.e. <code>on_draw_end</code>). The event listener then computes the geodesic length of the drawn geometry using the geometry service and prints it out:</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1142/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>